transistor/src/main/java/com/oneops/transistor/service/BomAsyncProcessor.java

```diff
@@ -102,6 +117,102 @@ public void compileEnv(long envId, String userId, Set<Long> excludePlats, CmsDep
         t.start();
     }
 
+    long reserveQuota(BomData bomData, String orgName, String userId) throws IOException {
+
+        Collection<CmsRfcCI> rfcCIs = bomData.getCis();
+        int totalCores = 0;
+        Map<String, Map<String, Integer>> quotaNeeded = new HashMap<>();
+
+        for (CmsRfcCI ciRfc : rfcCIs) {
+            String className = ciRfc.getCiClassName();
+            if (ciRfc.getRfcAction().equals("add")) {
```

Comment:
had a discussion with @okornev on this. We do not need to handle replace as of yet. Because [1 replace == 1 workorder]. So it is kind of a zero sum. We will tackle a very rare scenario where "delete" part of replace code succeeds but "add" fails